src/station_cmd.cpp

```diff
@@ -3802,6 +3804,13 @@ void IncreaseStats(Station *st, const Vehicle *front, StationID next_station_id)
 			 * among the wagons in that case.
 			 * As usage is not such an important figure anyway we just
 			 * ignore the additional cargo then.*/
+			Station *next_station = Station::Get(next_station_id);
+			if (current_order_is_implicit && !HasBit(GetAcceptanceMask(next_station), v->cargo_type)) {
+				/* Prevent node links to be created for implicit stops
+				 * whose stations do not accept the provided cargo. */
+				break;
+			}
+
 			IncreaseStats(st, v->cargo_type, next_station_id, v->refit_cap,
```

Comment:
As I read IncreaseStats, it records the capacity between to stations.
I think you want to change the "cargo demand at a station" in the flow planner.
This seems to be done in UpdateStationAcceptance(), but that already tests for station acceptance.

So, I do not understand how this issue of unloading stuff at C manifests, and how this fixes it.